public/app/features/explore/Logs/LogsTable.tsx

```diff
@@ -14,38 +15,42 @@ import {
   ValueLinkConfig,
 } from '@grafana/data';
 import { config } from '@grafana/runtime';
-import { Table } from '@grafana/ui';
+import { AdHocFilterItem, Table } from '@grafana/ui';
+import { FILTER_FOR_OPERATOR, FILTER_OUT_OPERATOR } from '@grafana/ui/src/components/Table/types';
 import { separateVisibleFields } from 'app/features/logs/components/logParser';
-import { parseLogsFrame } from 'app/features/logs/logsFrame';
+import { LogsFrame, parseLogsFrame } from 'app/features/logs/logsFrame';
 
 import { getFieldLinksForExplore } from '../utils/links';
 
+import { fieldNameMeta } from './LogsTableWrap';
+
 interface Props {
   logsFrames?: DataFrame[];
   width: number;
   timeZone: string;
   splitOpen: SplitOpen;
   range: TimeRange;
   logsSortOrder: LogsSortOrder;
+  columnsWithMeta: Record<string, fieldNameMeta>;
+  height: number;
+  onClickFilterLabel?: (key: string, value: string, refId?: string) => void;
+  onClickFilterOutLabel?: (key: string, value: string, refId?: string) => void;
+  datasourceType?: string;
 }
 
-const getTableHeight = memoizeOne((dataFrames: DataFrame[] | undefined) => {
-  const largestFrameLength = dataFrames?.reduce((length, frame) => {
-    return frame.length > length ? frame.length : length;
-  }, 0);
-  // from TableContainer.tsx
-  return Math.min(600, Math.max(largestFrameLength ?? 0 * 36, 300) + 40 + 46);
-});
-
-export const LogsTable: React.FunctionComponent<Props> = (props) => {
-  const { timeZone, splitOpen, range, logsSortOrder, width, logsFrames } = props;
-
+export function LogsTable(props: Props) {
+  const { timeZone, splitOpen, range, logsSortOrder, width, logsFrames, columnsWithMeta } = props;
   const [tableFrame, setTableFrame] = useState<DataFrame | undefined>(undefined);
 
+  // Only a single frame (query) is supported currently
+  const logFrameRaw = logsFrames ? logsFrames[0] : undefined;
+
   const prepareTableFrame = useCallback(
     (frame: DataFrame): DataFrame => {
-      const logsFrame = parseLogsFrame(frame);
+      // Parse the dataframe to a logFrame
+      const logsFrame = frame ? parseLogsFrame(frame) : undefined;
```

Comment:
Hm, not sure if `frame` will ever be falseish? If so, that should be respected in the type of this function as well.

Reply:
frame should never be falseish, but parseLogsFrame can return null, but the `frameWithOverrides` should always be a dataFrame, so the function always returns a dataFrame?

Reply:
Not sure I understand that, sorry. I understand that `parseLogsFrame` returns `DataFrame | null` but `frame` is never null, so you can just do:

```suggestion
      const logsFrame = parseLogsFrame(frame);
```

Reply:
Yeah my bad. Fixed!